### PR TITLE
Implement String.prototype.charAt()

### PR DESCRIPTION
**Pull request by lvidacs**
JerryScript-DCO-1.0-Signed-off-by: Laszlo Vidacs lvidacs.u-szeged@partner.samsung.com

### CODE REVIEW

File: tests/jerry/string-prototype-charat.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+// check negative
+assert("hello world!".charAt(-1) === "");
+
+assert("hello world!".charAt(-9999999) === "");
+
+assert("hello world!".charAt(-0) === "h");
+
+// check undefined
+assert("hello world!".charAt(undefined) === "h");
+
```
Comment (galpeter) [inline tests/jerry/string-prototype-charat.js]: Please add test case for boolean arguments also.

**Comment by egavrin** [comment]
Good to me.

**Comment by lvidacs** [comment]
Thanks for the review, boolean tests added.

**Comment by zherczeg** [comment]
LGTM

File: jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp [hunk 2]
```diff
@@ -112,7 +112,47 @@ static ecma_completion_value_t
...
+  /* 2 */
+  ECMA_TRY_CATCH (to_string_val,
+                  ecma_op_to_string (this_arg),
+                  ret_value);
+
+  /* 4 */
+  ecma_string_t *original_string_p = ecma_get_string_from_value (to_string_val);
+  const ecma_length_t len = ecma_string_get_length (original_string_p);
+
+  /* 3 */
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp]: Could you, please, clarify why step 4 is performed before step 3?

**Comment by lvidacs** [comment]
@ruben-ayrapetyan: The order is changed to follow the standard, thanks for the comment.

**Comment by egavrin** [comment]
`make push`

**Comment by galpeter** [comment]
Rebased & merged: https://github.com/Samsung/jerryscript/commit/632618da995b7bf7f5e04369c265405fb6837605

**Comment by egavrin** [comment]
@galpeter lets move to branches - since this PR wasn't automatically closed. Again.

**Comment by galpeter** [comment]
@egavrin we could move to branches but not all of us have rights to create branches.

**Pull request closed without merging**